test/integration/cloudflareCache.test.js

```diff
@@ -0,0 +1,200 @@
+      sinon.stub(logsQuery, "fetchLastAddedCacheLog").returns(cacheData.cacheModelData);
+      chai
+        .request(app)
+        .get("/cache")
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((error, response) => {
+          if (error) {
+            return done(error);
+          }
+
+          expect(response).to.have.status(200);
+          expect(response.body).to.be.an("object");
+          expect(response.body.message).to.be.a("string");
+          expect(response.body.count).to.be.a("number");
+          expect(response.body.timeLastCleared).to.be.a("number");
+          return done();
+        });
+    });
+
+    it("Should return latest purged cache metadata in last 24 hours", function (done) {
+      sinon.stub(logsQuery, "fetchCacheLogs").returns(cacheData.cacheModelData);
+      chai
+        .request(app)
+        .get("/cache")
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((error, response) => {
+          if (error) {
+            return done(error);
+          }
+
+          expect(response).to.have.status(200);
+          expect(response.body).to.be.an("object");
+          expect(response.body.message).to.be.a("string");
+          expect(response.body.count).to.be.a("number");
+          expect(response.body.timeLastCleared).to.be.a("number");
+
+          return done();
+        });
+    });
+
+    it("Should return unauthorized error when not logged in", function (done) {
+      chai
+        .request(app)
+        .get("/cache")
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(401);
+          expect(res.body).to.eql({
+            statusCode: 401,
+            error: "Unauthorized",
+            message: "Unauthenticated User",
+          });
+          return done();
+        });
+    });
+  });
+
+  describe("POST /cache", function () {
+    before(async function () {
+      await cleanDb();
+      await addUser(userData[0]);
+      const userId = await addUser(superUser);
+      jwt = authService.generateAuthToken({ userId });
+    });
+
+    beforeEach(async function () {
+      sinon.stub(cloudflare, "purgeCache").returns(cacheData.purgeCacheResponse[0]);
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+    });
+
+    it("Should purge the cache of member's profile page", function (done) {
+      chai
+        .request(app)
+        .post("/cache")
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Cache purged successfully");
+          expect(res.body.success).to.equal(true);
+          expect(res.body.errors).to.deep.equal([]);
+          expect(res.body.messages).to.deep.equal([]);
+          expect(res.body.result).to.be.a("object");
```

Comment:
why are we expecting empty arrays, i understand about errors to some extent but why do we need it in message ?

Reply:
It's the response returned from the Cloudflare API itself

Reply:
https://api.cloudflare.com/#zone-purge-files-by-url